java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2189.html

```diff
@@ -0,0 +1,47 @@
+<p>An infinite loop is one that will never end while the program is running, I.E. you have to kill the program to get out of the loop. Whether it is by meeting the loop's end condition or via a <code>break</code>, every loop should have an end condition.</p>
+
+<h2>Noncompliant Code Example</h2>
+<pre>
+for (;;) {  // Noncompliant; end condition omitted
+  // ...
+}
+
+for (int i = 0; i < 10; i--) { // Noncompliant; end condition but unreachedable
+  //...
+}
+
+int j;
+while (true) { // Noncompliant; end condition omitted
+  j++;
+}
+
+int k;
+boolean b = true;
+while (b) { // Noncompliant; b never written to in loop
+  k++;
+}
+</pre>
+
+<h2>Compliant Solution</h2>
+<pre>
+for (int i = 0; i < 10; i++) { // end condition now reachable
+  //...
+}
+
+int j;
+while (true) { // reachable end condition added
+  j++;
+  if (j  == Integer.MIN_VALUE) {  // true at Integer.MAX_VALUE +1
+    break;
+  }
+}
+
+int k;
+boolean b = true;
+while (b) {
+  k++;
+  b = k < Integer.MAX_VALUE;
+}
+</pre>
+<h2> See</h2>
+<li><a href="https://www.securecoding.cert.org/confluence/x/PYHfAw">CERT, MSC01-J.</a> - Do not use an empty infinite loop</li>
```

Comment:
List items should be between `<ul>` and `</ul>` tags...
Are you aware that rule descriptions can be generated?